### ISSUE TITLE
Undo (Ctrl+Z) triggers automatic snippet expansion

### ISSUE THREAD

**Issue by barro350** [draivin/hsnips#27]
Occasionally I want to revert the automatic snippet expansion using Ctrl+Z, but it won't work if the trigger contains regex or if it is with `A` flag. For example, consider the snippet

```javascript
snippet `([a-zA-Z])td` "superscript" iA
``rv = m[1] + '^{$1}'``$0
endsnippet
```

which expands `atd` to `a^{}`.

### Expected behavior

If I press Ctrl+Z, `a^{}` reverts back to `atd`.

### Actual behavior

If I press Ctrl+Z, first `a^{}` reverts back to `atd` but then is immediately and automatically converted to `a^{}` again.
So effectively Ctrl+Z achieves nothing.

### Additional Context

1. Undo works well if the snippet body has multiple lines:
```javascript
snippet `([a-zA-Z])td` "superscript" iA
newline
``rv = m[1] + '^{$1}'``$0
endsnippet
```
In this case Undo properly reverts the snippet expansion.

2. Even if the trigger does not contain regex, a similar problem exists.
```
snippet invs "inverse" A
^{-1}
endsnippet
```
`invs` is expanded to `^{-1}`. Pressing Ctrl+Z once converts it to `invs^{-1}`, and pressing Ctrl+Z once again reverts back to `invs`.

Overall, the interaction between Undo and automatic snippet expansion by `hsnips` is quite inconsistent.

**Comment by lancerXXXX** [comment]
I have the same problem, and on my vscode, it hasn't IntelliSense for hsnips after I update this extension to the latest version, do you have the same problem?

**Comment by draivin** [comment]
That's weird, I tried making it so it only triggers the automatic insertion when stuff is added, I'll look into it tomorrow!

**Comment by draivin** [comment]
This should be fixed on version 0.1.8, there is an edge case that happens when your trigger is a single character long, but in this case we're not able to tell the difference between a keypress and the undo action, so we are left with the previous behavior of retriggering the snippet.

**Comment by barro350** [comment]
Thank you for the quick fix.
Although I'm quite happy with version 0.1.8, I think it would be better if Ctrl+Z does not highlight the trigger text, just like default snippets of VS Code.
